WordPress/gutenberg
# Site Editor: Redesign table views

This creates a new Page component along with FilterBar and Table components. I've kept the design _mostly_ the same for this PR so that we can focus a follow up PR on updating design to reflect some of the more recent concepts, including the pattern management grid.

<img width="1512" alt="image" src="https://github.com/WordPress/gutenberg/assets/1072756/1bee63e7-9d13-49ee-885a-3bf2f10c9c19">

## Why?
These components are to be used by pages like the existing templates manage and the [upcoming library](https://github.com/WordPress/gutenberg/issues/50028) (replacing template parts). It's the first step towards resolving [this issue.](https://github.com/WordPress/gutenberg/issues/50430).

## How?
- Introduces a generic `Page` component that accepts title, subTitle, actions and children
- A FilterBar component which creates a set of filter inputs and filters an array of data
- A Table component with an API that closely mimics [TanStack Table](https://tanstack.com/table/v8/docs/examples/react/basic) which we can implement in future if needed
- Replaces the existing List component with a main router like component that renders either Template or Library (template_part) pages. I've separated these two due to the upcoming work around Library.

## Testing Instructions
- Open site editor
- Open templates or template parts
- Click manage
- Management views should work as expected

### Testing Instructions for Keyboard
